### PR TITLE
Avoid comparisons in python utilities when objects are not expected to be the same type

### PR DESCRIPTION
**Pull request by davidlange6**
#### PR description:

Avoid comparisons in python utilities when objects are not expected to be the same type. Also cleaned up unit test code to avoid warnings.

#### PR validation:

unit tests are ok. To be confirmed that there are no cmsrun differences

### CODE REVIEW

File: FWCore/ParameterSet/python/Config.py [hunk 1]
```diff
@@ -2379,6 +2378,8 @@ def testTask(self):
             # Also note that the mutating visitor replaces sequences and tasks that have
             # modified contents with their modified contents, it does not modify the sequence
             # or task itself.
+            print(process.path21.dumpPython(PrintOptions()))
+            print('cms.Path(process.mproducer10+process.mproducer8+process.mproducer8+(process.mproducer8+process.mproducer9)+process.sequence3, cms.Task(), cms.Task(process.None, process.mproducer10), cms.Task(process.d, process.mesproducer, process.messource, process.mfilter, process.mproducer10, process.mproducer2, process.myTask6), process.myTask100, process.myTask5)\n')
```
Comment (Dr15Jones) [inline FWCore/ParameterSet/python/Config.py]: I'd rather not have the test logs have print statements.

File: FWCore/ParameterSet/python/Config.py [hunk 2]
```diff
@@ -2831,6 +2832,8 @@ def testSwitchProducer(self):
             self.assertEqual((True,"EDProducer"), p.values["sp"][1].values["@module_edm_type"])
             self.assertEqual((True, "SwitchProducer"), p.values["sp"][1].values["@module_type"])
             self.assertEqual((True, "sp"), p.values["sp"][1].values["@module_label"])
+            print("look")
+            print(p.values["sp"])
```
Comment (Dr15Jones) [inline FWCore/ParameterSet/python/Config.py]: Please remove print statements

**Review by Dr15Jones — CHANGES_REQUESTED** [review]

File: FWCore/ParameterSet/python/Config.py (hunk 1 above)
Comment (davidlange6) [reply FWCore/ParameterSet/python/Config.py]: ah - that's a mistake!

**Comment by Dr15Jones** [comment]
please test

**Comment by smuzaffar** [comment]
please test

**Comment by fabiocos** [comment]
please test

**Comment by Dr15Jones** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**